Tasks/UseDotNetV2/usedotnet.ts

```diff
@@ -2,52 +2,84 @@
 import * as path from 'path';
 
 import * as tl from 'azure-pipelines-task-lib/task';
-
 import { DotNetCoreVersionFetcher } from "./versionfetcher";
+import { globalJsonFetcher } from "./globaljsonfetcher";
 import { VersionInstaller } from "./versioninstaller";
 import { Constants } from "./versionutilities";
 import { VersionInfo, VersionParts } from "./models"
 import { NuGetInstaller } from "./nugetinstaller";
 
 async function run() {
-    let packageType = tl.getInput('packageType') || "sdk";
+    let useGlobalJson: boolean = tl.getBoolInput('useGlobalJson');
     let versionSpec = tl.getInput('version');
-    if (versionSpec) {
-        console.log(tl.loc("ToolToInstall", packageType, versionSpec));
+    const nugetVersion = tl.getInput('nugetVersion') || '4.4.1';
+
+    // Check if we want install dotnet
+    if (versionSpec || useGlobalJson) {
+        let packageType = tl.getInput('packageType') || "sdk";
         let installationPath = tl.getInput('installationPath');
-        if (!installationPath) {
-            installationPath = path.join(tl.getVariable('Agent.ToolsDirectory'), "dotnet");
-        }
         let includePreviewVersions: boolean = tl.getBoolInput('includePreviewVersions');
+        let workingDirectory: string | null = tl.getPathInput("workingDirectory", false) || null;
```

Comment:
because typescript doesn't know that this can be null. I don't know why. The code behind the `= ` defines more than clear the workingDirectory can be null.